src/emu/device.cpp

```diff
@@ -217,11 +217,17 @@ std::string device_t::parameter(const char *tag) const
 
 void device_t::static_set_clock(device_t &device, u32 clock)
 {
+	device.m_derived_clock_dividend = 0;
+	device.m_derived_clock_divisor = 0;
+
 	// derive the clock from our owner if requested
 	if ((clock & 0xff000000) == 0xff000000)
 	{
+		device.m_derived_clock_dividend = (clock >> 12) & 0xfff;
+		device.m_derived_clock_divisor = (clock >> 0) & 0xfff;
+
 		assert(device.m_owner != nullptr);
-		clock = device.m_owner->m_configured_clock * ((clock >> 12) & 0xfff) / ((clock >> 0) & 0xfff);
+		clock = device.m_owner->m_configured_clock * device.m_derived_clock_dividend * device.m_derived_clock_divisor;
```

Comment:
You're right.  The only reason that this wasn't an issue for me was because:
1. In CoCo-land, both numbers are '1'
2. In my test scenario, the SAM was written to forcing a change that corrected the issue